src/main/db_main.cpp

```diff
@@ -0,0 +1,90 @@
+#include "main/db_main.h"
+#include <memory>
+#include "loggers/catalog_logger.h"
+#include "loggers/index_logger.h"
+#include "loggers/network_logger.h"
+#include "loggers/parser_logger.h"
+#include "loggers/settings_logger.h"
+#include "loggers/storage_logger.h"
+#include "loggers/transaction_logger.h"
+#include "settings/settings_manager.h"
+
+namespace terrier {
+void DBMain::InitLoggers() {
+  try {
+    init_main_logger();
+    // initialize namespace specific loggers
+    storage::init_index_logger();
+    storage::init_storage_logger();
+    transaction::init_transaction_logger();
+    catalog::init_catalog_logger();
+    settings::init_settings_logger();
+    parser::init_parser_logger();
+    network::init_network_logger();
+    // Flush all *registered* loggers using a worker thread.
+    // Registered loggers must be thread safe for this to work correctly
+    spdlog::flush_every(std::chrono::seconds(DEBUG_LOG_FLUSH_INTERVAL));
+  } catch (const spdlog::spdlog_ex &ex) {
+    std::cerr << "debug log init failed " << ex.what() << std::endl;  // NOLINT
+    throw ex;
+  }
+  // log init now complete
+  LOG_TRACE("Logger initialization complete");
+}
+
+void DBMain::Init() {
+  InitLoggers();
+
+  // initialize stat registry
+  main_stat_reg_ = std::make_shared<common::StatisticsRegistry>();
+
+  // create the global transaction mgr
+  auto *buffer_pool = new storage::RecordBufferSegmentPool(
+      type::TransientValuePeeker::PeekInteger(param_map_.find(settings::Param::buffer_pool_size)->second.value_),
+      10000);
```

Comment:
The size limit is via a setting, but the re-use limit is hard coded? IMO the latter should also be a settings value.